vignettes/Liwen.Rmd

```diff
@@ -0,0 +1,65 @@
+In this function, I use STRING score as an indicator of "related",if edges between two symbols does not exist in our database, we will ignore it and doesn't count in the averaging process.
+
+## Input Info
+Our input is just the curated systems information, a usable STRING database and a gene name (HGNC) of interest.
+
+## Function description
+1. Sum up the total score between each component gene in current system and the gene of interest
+2. Devide the total score with the number of components in each system to get final score
+3. Print information gained in each system
+4. Return the system with the highest final score among all systems
+
+There are several situations I considered: 
+
+* If there is no or very rare interaction between any component and the input gene, the final score should be (around) 0.
+* If System A and B have the same total number of components and most components in System A have higher interaction score than components in B, function should output system A.
+* If system A have lots of component, but only small portion of them have interactions with gene in interest, System B have a smaller amount of component, but with higher portion of genes which have interactions,(but smaller in absolute number). I want the function to output system B instead of system A
+
+So I choose to use [finalscore = totalscore/ #of components in the system] to capture the situations above.
+
+## Function Usage
+```
+#load data
+myURL <- paste0("http://steipe.biochemistry.utoronto.ca/abc/assets/",
+                "STRINGedges-2019-03-14.RData")
+load(url(myURL))
+myDB <- fetchData("SysDB")
+#use the function
+mostRelSys(myDB,"CFTR")
+#> mostRelSys(myDB,"CFTR")
+#total interaction found in system  PHALY is  4 with final score of  56.23077 .
+#total interaction found in system  SLIGR is  0 with final score of  0 .
+#total interaction found in system  NLRIN is  0 with final score of  0 .
+#total interaction found in system  HVGCR is  6 with final score of  263.9048 .
+#[1] "HVGCR"
+```
+
+## Function Insight
+I was trying to use this function to build a hierarchy between systems; suppose A is a subsystem of B,which means (almost) all components in A are in B,then for each gene A and B in common, most gene should give system A as an result of this function. 
+This function is more useful to generate hypothesis of the hierarchy, instead of a definitive answer, since the result describe above doesn't imply they have a subset relationship. But we can use it to filter some information out.
+
+## Problems
+One problem with this function and the hierachy hypothesis is the Running time, suppose among n system, each with m components, to generate hypothesis about hierarchy, we need to O(n*m)! to get the result, this is just ridiculusly large.
```

Comment:
While run time is a good concern to have, O(n*m) is not necessarily too large for an efficient algorithm. It seems to me this computation amounts to a few precomputations and a sum over a subset ... that should be quite doable on a genome scale.